apps/root/src/services/accountService.ts

```diff
@@ -376,218 +220,307 @@ export default class AccountService extends EventsManager<AccountServiceData> {
-      (!forceAddressMatch || address === this.user.signature.signer)
-    ) {
-      return this.user.signature;
-    }
+    const accounts = accountsResponse.accounts;
 
-    const adminWallets = this.user.wallets.filter(
-      (wallet) => wallet.status === WalletStatus.connected && wallet.isAuth
-    );
+    if (accounts.length) {
+      let connectedWallets: Wallet[] = [...(wallet ? [wallet] : [])];
 
-    const addressToUse = address || (adminWallets.length && adminWallets[0].address) || undefined;
+      for (const walletConnector of connectors || []) {
+        try {
+          const { address, providerInfo, walletClient } = await timeoutPromise(getConnectorData(walletConnector), '1s');
 
-    const lastSignatureRaw = localStorage.getItem(WALLET_SIGNATURE_KEY);
-    let signature;
-    if (lastSignatureRaw) {
-      const lastSignature = JSON.parse(lastSignatureRaw) as { id: Address; expiration: string; message: string };
+          connectedWallets.push(
+            toWallet({
+              address,
+              status: WalletStatus.connected,
+              walletClient: walletClient,
+              providerInfo: providerInfo,
+              isAuth: true,
+            })
+          );
+        } catch (e) {
+          console.error('Failed to parse wallet connector', walletConnector, e);
+        }
+      }
 
-      const lastExpiration = new Date(lastSignature.expiration).getTime();
+      connectedWallets = uniqBy(connectedWallets, 'address');
+      const parsedWallets = accounts[0].wallets.map<Wallet>((accountWallet) => {
+        const foundWallet = connectedWallets.find(({ address }) => accountWallet.address.toLowerCase() === address);
+        if (foundWallet) {
+          return {
+            ...foundWallet,
+            isAuth: accountWallet.isAuth,
+          };
+        }
+
+        return toWallet({
+          address: accountWallet.address,
+          isAuth: accountWallet.isAuth,
+          status: WalletStatus.disconnected,
+        });
+      });
 
-      const adminWalletsAddresses = forceAddressMatch ? [address] : adminWallets.map((wallet) => wallet.address);
-      if (adminWalletsAddresses.includes(lastSignature.id) && today < lastExpiration) {
-        signature = {
-          expiration: lastSignature.expiration,
-          message: lastSignature.message,
-          signer: lastSignature.id,
-        };
+      const walletIsInUser = find(parsedWallets, ({ address }) => address === wallet?.address);
+
+      this.accounts = accounts;
+      this.user = {
+        id: this.accounts[0].id,
+        label: this.accounts[0].label,
+        status: UserStatus.loggedIn,
+        wallets: parsedWallets,
+        signature: storedSignature,
+      };
+
+      if (wallet && walletIsInUser) {
+        await this.setActiveWallet(wallet.address);
       }
+    } else {
+      await this.createUser({ label: 'Personal', signature: storedSignature, wallet });
```

Comment:
Since the user is not changed untill the last update, we are not going to be fetching untill we actually create the user, so the api the is going to respond, just empty :D